include/fuse_common.h

```diff
@@ -32,14 +32,22 @@ extern "C" {
 #endif
 
 /**
- * Information about open files
+ * Information about an open file handle
```

Comment:
Let's keep "open files" here. Filesystems aren't obliged to put anything into fi->fh, in which case there is no file handle (yet there is a fuse_file_info structure).

Reply:
We seem to be using the word "file handle" to refer to two different things.  There's the 64-bit fh field, of course.  But there's also the thing that you create during `open`.  For every `open` there will be exactly one `release`, which implies that both client and server must be tracking whatever `open` created (even if the server always sets the same value for `fh`).  If that's not a file handle, then what is it?

Reply:
That probably depends on which kernel you use libfuse with. I would say on the libfuse side there is no need to worry about that.

Reply:
Kernel aside, there are three distinct things in libfuse that currently share two names:
1) The thing identified by a node_id, called a "file"
2) The thing identified by a node_id and a file handle, created by `open`, `opendir`, or `create`, and described by `fuse_file_info`.  What should it be called?  You've previously said this should be called a "file handle" (https://github.com/libfuse/libfuse/issues/389#issuecomment-478732346).  Are you having second thoughts?
3) The 64-bit value stored in `fuse_file_info.fh` variously called "file handle" or "file handle id".  The latter makes more sense to me if thing 2 will be called a "file handle".

Reply:
Hmm. Yeah, this is messy. Here's my take on it (from general principles without looking at which terms are actually used right now):

(1) should not be a called a file (since it may be a directory). I would refer to it as a "directory entry". 

I would not say that (2) is a thing at all. I would say that `fuse_file_info` provides information about either an "open file" or an "open directory". However, most filesystems will create some internal structure to keep track of open files or directories that libfuse knows nothing about - that is the "file handle".

(3) is the "file handle id"